### ISSUE TITLE
Top-level await does not seem to work in Meteor 3

### ISSUE THREAD

**Issue by alisnic** [meteor/meteor#12986]
Given I have a file with the following contents:

```js
import axios from "axios";

const getAndrei = async () => {
  const response = await axios.get('https://api.github.com/users/alisnic')
  return response.data
};

const Andrei = await getAndrei();

export default Andrei;
```

And then I import it:

```js
import Andrei from './tla.js'

console.log('imported', Andrei)
```

## Expected behaviour

json data from the response is logged

## Actual behaviour

The output is `imported { [Symbol(__esModule)]: true }`. 

In Meteor 2, everything works. Repo with repro - https://github.com/alisnic/meteor3-tla-repro (do a npm start to see the log).

### meteor/packages

```
meteor-base@1.5.2-beta300.0             # Packages every Meteor app needs to have
mobile-experience@1.1.1-beta300.0       # Packages for a great mobile UX
mongo@2.0.0-beta300.0                   # The database Meteor supports right now
reactive-var@1.0.13-beta300.0            # Reactive variable for tracker

standard-minifier-css@1.9.3-beta300.0   # CSS minifier run for production mode
standard-minifier-js@3.0.0-beta300.0    # JS minifier run for production mode
es5-shim@4.8.1-beta300.0                # ECMAScript 5 compatibility for older browsers
ecmascript@0.16.8-beta300.0              # Enable ECMAScript2015+ syntax in app code
typescript@4.9.5-beta300.0              # Enable TypeScript syntax in .ts and .tsx modules
shell-server@0.6.0-beta300.0            # Server-side component of the `meteor shell` command
hot-module-replacement@0.5.4-beta300.0  # Update client in development without reloading the page
```

**Comment by alisnic** [comment]
@zodern maybe you can shed some light here? what am I doing wrong?

**Comment by jbo023** [comment]
hmm, we have a similar problem, with Meteor 3 beta 4

```
collection.ts

export const Roles = await getRolesCollection();
```
```
model.ts 
import {
  Roles,
} from './collections';
console.log('rolesModel', Roles);
```

Roles here is undefined.

**Comment by perbergland** [comment]
New simpler repro repo:

https://github.com/perbergland/tla-repro/tree/main

The produced reified code for tla-file.js does NOT look correct. It differs from the example in the reify readme file since it is not marked as async: true

```build/programs/server/app/app.js

/////////////////////////////////////////////////////////////////////////////////////////////
//                                                                                         //
// server/tla-file.js                                                                      //
//                                                                                         //
/////////////////////////////////////////////////////////////////////////////////////////////
                                                                                           //
!module.wrapAsync(async function (module1, __reifyWaitForDeps__, __reify_async_result__) {
  "use strict";
  try {
    module1.export({
      tlaValue: () => tlaValue,
      someFunction: () => someFunction
    });
    let axios;
    module1.link("axios", {
      default(v) {
        axios = v;
      }
    }, 0);
    if (__reifyWaitForDeps__()) (await __reifyWaitForDeps__())();
    const asyncMethod = async () => {
      return 4;
    };
    console.log("before tla init");
    const tlaValue = await asyncMethod();
    console.log("after tla init");
    const someFunction = () => {
      return 7;
    };
    __reify_async_result__();
  } catch (_reifyError) {
    return __reify_async_result__(_reifyError);
  }
  __reify_async_result__()
}, {
  self: this,
  async: false
});
``

**Comment by perbergland** [comment]
This is a bug in https://github.com/meteor/reify - it does not handle the case where the await is in the right hand side of a top level assignment, e.g. 

```
const x = await functionCall();
```

I would rate this as a top level priority to fix for Meteor 3
https://github.com/meteor/reify does not have a separate issue list, so this is as good place as any to track the bug.

Workaround:
Add a top level await call in the imported file, e.g. 

```
await 0;
```

anywhere in the file

**Comment by perbergland** [comment]
The code to fix (along with a new test case) is:

https://github.com/meteor/reify/blob/444ecd862f4a9d5da02b355bc49ccc63ab97e2b3/lib/import-export-visitor.js#L389-L402

**Comment by perbergland** [comment]
Thanks to @zodern for the pointer to the reify code that needs fixing

**Comment by leonardoventurini** [comment]
I am taking a look at the fix [here](https://github.com/meteor/reify/pull/6) and checking if I can expand on that to cover a wider range of cases.

**Comment by leonardoventurini** [comment]
This was fixed here, forgot to link the issue I think

https://github.com/meteor/reify/pull/8

**Comment by Torgen** [comment]
Is this believed to be fixed in 3.0.1? I'm still having this issue in a test that imports a top-level-await module, but at least the workaround of awaiting without assigning the result works for me.

**Comment by perbergland** [comment]
I can confirm that this is NOT working in meteor 3.0.1 but I cannot figure out why. The workaround is still needed.
I checked the meteor version contents and the correct version of reify (0.25.2) is brought in.

Filing an issue in the meteor/reify repo

**Comment by denihs** [comment]
I've tested this with the PR https://github.com/meteor/meteor/pull/13298, and it worked as expected. 

After the tests pass, we will merge it, and it will be included in version 3.0.4.
